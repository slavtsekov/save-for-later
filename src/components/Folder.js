import React from 'react';

const Folder = (props) => (
    <div>
        <p>{props.text}</p>
        <button
            type="button"
            onClick={(ะต) => {
                props.onDelete(props.text);
            }}
        >
            Remove
        </button>
    </div>
);

export { Folder as default };
