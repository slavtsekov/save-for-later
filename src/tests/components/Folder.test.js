import React from 'react';
import { shallow } from 'enzyme';
import Folder from '../../components/Folder';

test('should render Folder correctly', () => {
    const wrapper = shallow(<Folder text="Default" />);
    expect(wrapper).toMatchSnapshot();
});

test('should call onDelete when button is clicked', () => {
    const onDeleteMock = jest.fn();
    const wrapper = shallow(<Folder text="Default" onDelete={onDeleteMock} />);

    wrapper.find('button').simulate('click');

    expect(onDeleteMock).toHaveBeenCalled();
});
