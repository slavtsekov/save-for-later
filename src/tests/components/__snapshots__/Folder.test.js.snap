// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Folder correctly 1`] = `
<div>
  <p>
    Default
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</div>
`;
